ui/src/containers/Award/component/AwardList.tsx

```diff
@@ -1,16 +1,32 @@
-import React from 'react'
+import React, { useMemo } from 'react'
 import { useQuery } from '@apollo/client'
 
 import style from './AwardListStyle.scss'
 import logo from '../../../assets/access-logo.png'
 import AwardItem from './AwardItem'
 import { AWARD_LIST_QUERY } from '../../../query/award'
+import { CANDIDATE_LIST_QUERY } from '../../../query/candidates'
 import { AwardsList } from '../../../__generated__/AwardsList'
+import { CandidatesList } from '../../../__generated__/CandidatesList'
 import AwardCreation from './AwardCreation'
+import AwardDerived from './AwardDerived'
 
 function AwardList() {
   const { data: awardsListData } = useQuery<AwardsList>(AWARD_LIST_QUERY)
-
+  const { data: candidatesListData } = useQuery<CandidatesList>(CANDIDATE_LIST_QUERY)
+  const awardsData: AwardDerived[] = useMemo(
+    () =>
+      awardsListData?.awards?.map((award: any) => {
+        const candidateName =
+          candidatesListData?.candidates?.find((candidateToCheck: any) => candidateToCheck.id === award.winner)?.name ??
+          'unknown'
+        return {
+          ...award,
+          winnerName: candidateName
+        }
+      }) ?? [],
+    [awardsListData, candidatesListData]
+  )
```

Comment:
just a suggestion~ imho, rather than 'manually' mapping winner and awards, assigning winner with `candidate` type will be simpler and easier. can fully use the advantage of graphql.

Reply:
good point, can you show me for this? This is my very first time use gql. I struggle a lot